registry/storage/driver/middleware/cloudfront/middleware.go

```diff
@@ -124,6 +156,32 @@ func (lh *cloudFrontStorageMiddleware) URLFor(ctx context.Context, path string,
 		return lh.StorageDriver.URLFor(ctx, path, options)
 	}
 
+	if lh.awsIPs != nil {
+		if addr, err := s3filter.ParseIPFromRequest(ctx); err == nil {
```

Comment:
All these `else` statements are a little messy. It might be better to define a helper method to handle this.